product_configurator_wizard/wizard/product_configurator.py

```diff
@@ -805,14 +805,17 @@ def action_config_done(self):
         if self.product_id:
             remove_cv_links = map(lambda cv: (2, cv), self.product_id.value_custom_ids.ids)
             new_cv_links = self.product_id.product_tmpl_id.encode_custom_values(custom_vals)
-            self.product_id.write({
-                'attribute_value_ids': [(6, 0, self.value_ids.ids)],
-                'value_custom_ids':  remove_cv_links + new_cv_links,
-            })
-            if self.order_line_id:
-                self.order_line_id.write(self._extra_line_values(self.order_line_id.order_id,
-                                                                 self.product_id,
-                                                                 new=False))
+
+            if self.product_id.attribute_value_ids.ids != self.value_ids.ids:
+                new_product = self.product_tmpl_id.create_variant(self.value_ids.ids, custom_vals)
+                self.order_line_id.write({'product_id': new_product.id})
```

Comment:
This indeed does make sense, if the attribute values are changed, there is a chance another variant exists with the same attr values.

However, the hook introduced by Richard "_extra_line_values" should be kept.

Another consideration would be changing the create_variant method name to get_variant or search_create_variant as this implies returning either an existing or a new variant. (To avoid confusion).

Lastly if a custom_value is searchable that means it should also generate a new variant but it's a but more advanced.

If you can handle the first two that would be great, if not i will do them myself.